ddl/table.go

```diff
@@ -53,29 +55,65 @@ func onCreateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 		}
 	}
 
-	ver, err = updateSchemaVersion(t, job)
-	if err != nil {
-		return ver, errors.Trace(err)
-	}
-
+	originalState := job.SchemaState
 	switch tbInfo.State {
 	case model.StateNone:
 		// none -> public
-		tbInfo.State = model.StatePublic
+		if withSelect {
+			tbInfo.State = model.StateWriteOnly
+		} else {
+			tbInfo.State = model.StatePublic
+		}
 		tbInfo.UpdateTS = t.StartTS
 		err = t.CreateTable(schemaID, tbInfo)
 		if err != nil {
+			job.State = model.JobStateCancelled
 			return ver, errors.Trace(err)
 		}
 		if EnableSplitTableRegion {
 			// TODO: Add restrictions to this operation.
 			go splitTableRegion(d.store, tbInfo.ID)
 		}
+		ver, err = updateVersionAndTableInfo(t, job, tbInfo, originalState != tbInfo.State)
+		if err != nil {
+			job.State = model.JobStateCancelled
+			return ver, err
+		}
 		// Finish this job.
-		job.FinishTableJob(model.JobStateDone, model.StatePublic, ver, tbInfo)
+		job.FinishTableJob(model.JobStateDone, tbInfo.State, ver, tbInfo)
```

Comment:
If withSelect is true, we can't finish this job, if the job state is model.JobStateDone, we will dequeue the job from the ddl job list, and put it into history job list, then you will execute the insertion in DDLExec, but actually, the job is finished, if you crash in the DDLExec before finishing inserting, then we don't know the job is failed.

Reply:
And when state is model.JobStateDone, TiDB will write out binlog, it is also a side effect.